bentoml/_internal/frameworks/keras.py

```diff
@@ -188,43 +191,38 @@ def __init__(
         batch_options: t.Optional[t.Dict[str, t.Any]],
         model_store: "ModelStore" = Provide[BentoMLContainer.model_store],
     ):
+        in_store_tag = model_store.get(tag).tag
         super().__init__(
-            tag=tag,
+            tag=str(in_store_tag),
             predict_fn_name=predict_fn_name,
             device_id=device_id,
             partial_kwargs=predict_kwargs,
             resource_quota=resource_quota,
             batch_options=batch_options,
             model_store=model_store,
         )
-        self._session = get_session()
-
-    @property
-    def session(self) -> "BaseSession":
-        return self._session
+        self._session = _sess
+        self._tag = in_store_tag
 
-    # pylint: disable=attribute-defined-outside-init
-    def _setup(self) -> None:
+    # pylint: disable=arguments-differ,attribute-defined-outside-init
+    def _setup(self) -> None:  # type: ignore[override] # noqa
```

Comment:
why do we need to do this? `_setup` is not user facing and not supposed to accept any extra arguments